### DISCUSSION TITLE
Camo cell

### DISCUSSION THREAD

**Discussion by adamorgler** [MaxRobinsonTheGreat/EvolutionSimulatorV2#5] [Ideas and Suggestions]
I added a camo cell into my fork of the game. When observed, all cells beyond the camo cell are invisible to the observer. Currently, the camo cell is also invisible. To balance this, you could implement that eye's can directly observe the camo cells, partially negating their usefulness.  I haven't added that but a simple edit to the Brain class should do the trick.

![image](https://user-images.githubusercontent.com/47910544/141753614-0bbe242a-e496-4966-8fb6-aa725137f364.png)
Here is an example of a creature that is completely 'invisible' to any outside observer. The dark green cells are the camo cells.  I ran a few tests of creatures that move towards producer cells and they never saw this creature. (I made sure to test against a visible producer.)

I have yet to observe how camo cells will potentially naturally evolve but there is some interesting potential here.

I will submit a pull request so if you wish to add this to your simulation, you have the option to do so.  Let me know if you have any questions! 😄

**Comment by MaxRobinsonTheGreat** [comment]
This is a very neat idea, and I have thought a lot about implementing camo a few times. Thanks for taking the initiative and writing some code! 

Your implementation of the camo cell basically works by pretending to be an empty cell, which can be used to hide from predators. I had not thought of this, and I think it's an interesting idea. 

However, I've run your code a few times for a good while and don't notice the orgs using the camo cells very much. I think that they are ultimately not nearly as useful as other cells. For instance, an org could evolve a red cell instead which would scare AND kill predators, not just hide from them. 

Another implementation would be that camo cells can pretend to be any cell. Unfortunately this has the same problem as before (why evolve a fake killer cell instead of an actual killer cell?). Perhaps pretending to be food could lure in prey, but I think this is the only useful case. Maybe it should be a lure cell? 

Really appreciate the input and code, and I think with more development something like this can be added in the future 😁

**Reply by adamorgler** [reply]
I actually tried implementing the camo cell imitating other cells but ran into issues with rendering the different colors. I'll have to go back and find a way to implement this and see how it evolves.

I also noticed the camo cell has trouble evolving. I ran some tests of different camouflaged creatures and found some interesting results. Moving creatures with the camo quickly evolved those cells away as they had little use. Plants actually used the camo very well and would survive a long time if they could quickly group together into clusters. When I had a camo creature vs a killer creature, they would actually stay on their respective sides of the map and rarely interact ( The killers would run towards the camo creatures but once the camo creature saw the killer is would turn an run, showing its camo cells. The killer would then in turn lose sight and give up the chase.). However, all these tests were with engineered creatures.

The camo only seemed useful when used in larger structures which the sim has trouble evolving.

**Reply by doo158** [reply]
well really i think armor has a similar issue, with it being difficult to evolve naturally and being useful in engineered creatures
